pkg/contexts/ocm/transfer/transfer.go

```
@@ -124,65 +139,93 @@ func CopyVersion(printer common.Printer, log logging.Logger, hist common.History
 	*t.GetDescriptor() = *src.GetDescriptor().Copy()
 	log.Info("  transferring resources")
 	for i, r := range src.GetResources() {
-		var m ocm.AccessMethod
-		a, err := r.Access()
-		if err == nil {
-			m, err = r.AccessMethod()
-			if err == nil {
-				defer m.Close()
-				ok := a.IsLocal(src.GetContext())
-				if !ok {
-					if !none.IsNone(a.GetKind()) {
-						ok, err = handler.TransferResource(src, a, r)
-						if !ok {
-							log.Info("transport omitted", "resource", r.Meta().Name, "index", i, "access", a.GetType())
-						}
+		c := &runContext{}
+
+		// gather the access and the access method
+		a := run(c, r.Access)
+		m := run(c, r.AccessMethod)
+
+		if c.err == nil {
```

why not start by failing early like:

```go
		if c.err != nil {
			if m != nil {
				m.Close()
			}

			if !errors.IsErrUnknownKind(c.err, errors.KIND_ACCESSMETHOD) {
				return errors.Wrapf(c.err, "%s: transferring resource %d", hist, i)
			}

			printer.Printf("WARN: %s: transferring resource %d: %s (enforce transport by reference)\n", hist, i, c.err)
		} else {
			...
		}
```